include/utils/linearAlg2D.h

```diff
@@ -67,6 +67,20 @@ class LinearAlg2D
         return -1;
     }
 
+    /*!
+     * A single-shot line-segment/line-segment intersection that returns the parameters and doesn't require a grid-calculation beforehand.
+     *
+     * \param p1 The start point of the first line segment.
+     * \param p2 The end point of the first line segment.
+     * \param p3 The start point of the second line segment.
+     * \param p4 The end point of the second line segment.
+     * \param t The parameter of the intersection on the first line segment (intersection = p1 + t * (p2 - p1)).
+     * \param u The parameter of the intersection on the second line segment (intersection = p3 + u * (p4 - p3)).
+     *
+     * \return Whether the two line segments intersect.
+     */
+    static bool lineSegmentLineSegmentIntersection(const Point2LL& p1, const Point2LL& p2, const Point2LL& p3, const Point2LL& p4, float* t, float* u);
```

Comment:
Couldn't this be merged with `lineLineIntersection`, or make one use the other ? I would say that in your implementation, you do calculate the lines intersection, then check whether the points are inside the segments.

Reply:
I've had a bit of a closer look at `lineLineIntersection`, and now I'm not sure why I did the other thing in the 1st place 😅 

I'll adapt this.

Reply:
done!